app/models/category.rb

```diff
@@ -647,22 +641,11 @@ def clear_auto_bump_cache!
   end
 
   def self.auto_bump_topic!
-    bumped = false
-
-    auto_bumps =
-      CategoryCustomField
-        .where(name: Category::NUM_AUTO_BUMP_DAILY)
-        .where('NULLIF(value, \'\')::int > 0')
-        .pluck(:category_id)
-
-    if (auto_bumps.length > 0)
-      auto_bumps.shuffle.each do |category_id|
-        bumped = Category.find_by(id: category_id)&.auto_bump_topic!
-        break if bumped
-      end
-    end
-
-    bumped
+    Category
+      .joins(:category_setting)
+      .where("category_settings.num_auto_bump_daily > 0")
+      .shuffle
+      .any?(&:auto_bump_topic!)
```

Comment:
Two changes led to the simplification of this method:

1. The query is much simpler for the sidecar table than for custom fields.
2. We can use the fact that enumerating an empty collection is a no-op, and the behaviour of `#any?` together with the fact that `#auto_bump_topic!` returns a boolean.

The behaviour is regression tested in the existing Category test.